### DISCUSSION TITLE
How to Prevent OpenVPN3 from Updating resolv.conf

### DISCUSSION THREAD

**Discussion by gponty** [OpenVPN/openvpn3-linux#267] [Q&A - Our support arena] [answered]
Hello,

I'm experiencing an issue with OpenVPN3 where it automatically updates the resolv.conf file on my system, which causes conflicts with my existing DNS settings.

I've tried several methods to prevent OpenVPN3 from modifying the resolv.conf file, including:

Adding custom scripts with the up and down options in the configuration file.
Using the chattr +i /etc/resolv.conf command to make the file immutable.
Experimenting with options like dhcp-option DOMAIN-ROUTE . to work with systemd-resolved.
Unfortunately, none of these approaches seem to work with OpenVPN3, as the resolv.conf file continues to get updated during the VPN connection process.

Is there a recommended way to prevent OpenVPN3 from modifying the DNS settings or a specific method that I might have overlooked?

Any help or suggestions would be greatly appreciated.

Thank you!

**Comment by dsommers** [comment, marked as answer]
Which version of OpenVPN 3 Linux do you run?  And which distro are you on?  As of `v22_dev` and `v23`, this should definitely be possible to do easily by this command:

     # openvpn3-admin netcfg-service --config-unset resolv-conf

See the [`openvpn3-admin-netcfg-service` man page](https://codeberg.org/OpenVPN/openvpn3-linux/src/branch/master/docs/man/openvpn3-admin-netcfg-service.8.rst) for details.  I believe this works also on `v20` and `v21`, but I encourage you to upgrade to the latest `v23` release regardless.

With that done, to enable using `systemd-resolved`, you need this command line:

     # openvpn3-admin netcfg-service --config-set systemd-resolved true

However, if `systemd-resolved` is not enabled by default on your system - you need to ensure that service is configured and started properly without a VPN connection first.  Then OpenVPN 3 Linux can hook into that one.  Using this approach will also give you proper split-dns out-of-the-box (see the [`openvpn3-config-manage` man page](https://codeberg.org/OpenVPN/openvpn3-linux/src/branch/master/docs/man/openvpn3-config-manage.1.rst) for details.

**Comment by gponty** [comment]
Thank you David, work like a charm !
